docs/design/spatial-index/gen-spatial-index-example.py

```
@@ -0,0 +1,130 @@
+C_NOTE = "#74808B"       # Carbon 600, for the small legend note
+
+out = []
+out.append(f'<svg xmlns="http://www.w3.org/2000/svg" width="{W}" height="{H}" font-family="Helvetica,Arial,sans-serif">')
+out.append(f'<rect width="{W}" height="{H}" fill="#ffffff"/>')
+out.append(f'<text x="{MX}" y="34" font-size="20" font-weight="700" fill="{TXT}">Spatial index: covering cells for three geometries (domain [0,8)²)</text>')
+
+# --- grid: unit, then L3 (size2), L2 (size4), L1 (size8) on top ---
+def vline(x,color,w): out.append(f'<line x1="{sx(x)}" y1="{sy(0)}" x2="{sx(x)}" y2="{sy(8)}" stroke="{color}" stroke-width="{w}"/>')
+def hline(y,color,w): out.append(f'<line x1="{sx(0)}" y1="{sy(y)}" x2="{sx(8)}" y2="{sy(y)}" stroke="{color}" stroke-width="{w}"/>')
+for i in range(9):
+    vline(i, G_UNIT, 1); hline(i, G_UNIT, 1)
```

Ruff (0.15.18)

[error] Multiple statements on one line (semicolon)

(E702)